lib/kafka_ex.ex

```diff
@@ -253,18 +254,13 @@ Optional arguments(KeywordList)
   end
 
   @doc """
-  Returns a stream that consumes fetched messages.
-  This puts the specified worker in streaming mode and blocks the worker indefinitely.
-  The handler is a normal GenEvent handler so you can supply a custom handler, otherwise a default handler is used.
-
-  This function should be used with care as the queue is unbounded and can cause OOM.
-
+  Returns a stream that consumes fetched messages, the stream will halt once the max_bytes number of messages is reached, if you want to halt the stream early supply a small max_bytes, for the inverse supply a large max_bytes.
   Optional arguments(KeywordList)
+  - stream_mode: the mode the stream will be in, `:infinite` for an infinite stream and `:finite` for a finite stream, a finite stream will return once there's no data left to consume from the partition while an infinite stream will block till there's data to consume from the stream.
```

Comment:
I think it would be good if this mirrors the command line tools and uses something like `:no_wait_at_logend` https://cwiki.apache.org/confluence/display/KAFKA/System+Tools#SystemTools-SimpleConsumerShell